etcdctl/ctlv3/command/transfer_leadership_command.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/coreos/etcd/pkg/types"
+	"github.com/spf13/cobra"
+)
+
+// NewTransferLeadershipCommand returns the cobra command for "transfer-leadership".
+func NewTransferLeadershipCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "transfer-leadership <transferee>",
+		Short: "Transfers leadership.",
+		Run:   transferLeadershipCommandFunc,
+	}
+	return cmd
+}
+
+// transferLeadershipCommandFunc executes the "compaction" command.
+func transferLeadershipCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		ExitWithError(ExitBadArgs, fmt.Errorf("transfer-leadership command needs 1 argument"))
+	}
+
```

Comment:
can this figure out the current leader instead of expecting the user to supply the leader's endpoint?